File: shared/domain/place_repository.go
package domain

import "github.com/SeijiShii/home-visit-suite/shared/domain/models"

// PlaceRepository は場所データの永続化インターフェース。
type PlaceRepository interface {
	ListPlaces(areaID string) ([]models.Place, error)
	GetPlace(id string) (*models.Place, error)
	SavePlace(place *models.Place) error
	// DeletePlace は論理削除（DeletedAt をセット）。
	DeletePlace(id string) error
	// ListDeletedPlacesNear は指定座標から半径 radiusMeters メートル以内の削除済み場所を返す。
	ListDeletedPlacesNear(lat, lng, radiusMeters float64) ([]models.Place, error)
}

File: pwa/src/domain/repositories/place-repository.ts
// 場所データの永続化インターフェース。
// 参照実装: shared/domain/place_repository.go

import type { Place } from "../models/place";

export interface PlaceRepository {
  listPlaces(areaId: string): Promise<Place[]>;
  getPlace(id: string): Promise<Place | null>;
  savePlace(place: Place): Promise<void>;
  /** 論理削除（deletedAt をセット）。 */
  deletePlace(id: string): Promise<void>;
  /** 指定座標から半径 radiusMeters メートル以内の削除済み場所を返す。 */
  listDeletedPlacesNear(
    lat: number,
    lng: number,
    radiusMeters: number,
  ): Promise<Place[]>;
  /** 指定 Building 配下の論理削除済み部屋を返す（部屋の同番号復元用）。 */
  listDeletedRooms(buildingId: string): Promise<Place[]>;
}
